recipes/newrelic/infrastructure/darwin.yml

```diff
@@ -26,8 +25,19 @@ install:
   silent: true
 
   vars:
-    CONFIG_FILE: "/usr/local/etc/newrelic-infra/newrelic-infra.yml"
-    CONFIG_FILE_DIR: "/usr/local/etc/newrelic-infra"
+    # Variables prefixed with VN_ as they are parsed in alphabetical order, and they have dependencies
+    # HOMEBREW_PREFIX is different in intel/amd:
+    # intel: /usr/local
+    # amd: /opt/homebrew
+    V1_HOMEBREW_INSTALL_DIR:
+      sh: |
+        if [ $( sudo -i -u $SUDO_USER which brew ) ];then
+          sudo -i -u $SUDO_USER brew config | grep HOMEBREW_PREFIX | awk '{print $2}'
+        fi
+    V2_CONFIG_FILE:
+      sh: echo {{.V1_HOMEBREW_INSTALL_DIR}}/etc/newrelic-infra/newrelic-infra.yml
+    V2_CONFIG_FILE_DIR:
+      sh: echo {{.V1_HOMEBREW_INSTALL_DIR}}/etc/newrelic-infra
```

Comment:
@amolero-nr I had some trouble with the variables ☝️  as they have dependencies and seem to be parsed in alphabetical order. Is there a "better" way to do this? Thanks!

Reply:
Hi @rubenruizdegauna! Since I did not have a concrete answer :), I had to try `task` locally in my mac with this below and seemed to work fine:

```
# Taskfile.dev
version: '3'

vars:
  HOMEBREW_DIR: 
    sh: brew config | grep HOMEBREW_PREFIX | awk '{print $2}'
  CONFIG_FILE:
    sh: echo "{{.HOMEBREW_DIR}}/etc/newrelic-infra/newrelic-infra.yml"
  CONFIG_DIR:
    sh: echo "{{.HOMEBREW_DIR}}/etc/newrelic-infra"

tasks:
  default:
    cmds: 
      - |
        echo "HOMEBREW_DIR: {{.HOMEBREW_DIR}}"
        echo "CONFIG_FILE: {{.CONFIG_FILE}}"
        echo "CONFIG_DIR: {{.CONFIG_DIR}}"
    silent: true
```

And ran it like this:
```
task
```

```
# Ouput
amolero taskfile % task 
HOMEBREW_DIR: /opt/homebrew
CONFIG_FILE: /opt/homebrew/etc/newrelic-infra/newrelic-infra.yml
CONFIG_DIR: /opt/homebrew/etc/newrelic-infra
```

Also, I think it is better to move the check for the presence of the `brew` command into a `requireAtDiscovery` step inside the recipe, similar to [this one](https://github.com/newrelic/open-install-library/blob/main/recipes/newrelic/infrastructure/ohi/mysql/rhel.yml#L37), for example, and exit early if not found. Then, your `HOMEBREW_DIR` env var setup should be simpler.

cc: @NRhzhao,@Julien4218 (in case I missed something).

Reply:
Thanks @amolero-nr , I'll take a look. For some reason when running the recipe directly from newrelic-cli I couldn't manage to make the variables work 🤷‍♂️

Reply:
Hi @amolero-nr , I tried the task command and it works, but for some reason when I run the recipe with newrelic-cli it does not :( 

I built newrelic-cli and executed it locally:
```
sudo NEW_RELIC_API_KEY=XXXXXX NEW_RELIC_ACCOUNT_ID=XXXXXXXX NEW_RELIC_REGION=XXXXX ./bin/darwin/newrelic install --debug    -c ../open-install-library/recipes/newrelic/infrastructure/darwin.yml 
```
Output:
```
...
DEBUG executing recipe infrastructure-agent-installer 
DEBUG /etc/newrelic-infra/newrelic-infra.yml       
DEBUG /opt/homebrew                                
DEBUG /etc/newrelic-infra                          
DEBUG Task execution returned error    
....
```

And changing `HOMEBREW_INSTALL_DIR` to `A_HOMEBREW_INSTALL_DIR` works:
```  
vars:
    # Variables prefixed with VN_ as they are parsed in alphabetical order, and they have dependencies
    # HOMEBREW_PREFIX is different in intel/amd:
    # intel: /usr/local
    # amd: /opt/homebrew
    A_HOMEBREW_INSTALL_DIR:
      sh: |
        if [ $( sudo -i -u $SUDO_USER which brew ) ];then
          sudo -i -u $SUDO_USER brew config | grep HOMEBREW_PREFIX | awk '{print $2}'
        fi
    CONFIG_FILE:
      sh: echo {{.A_HOMEBREW_INSTALL_DIR}}/etc/newrelic-infra/newrelic-infra.yml
    CONFIG_FILE_DIR:
      sh: echo {{.A_HOMEBREW_INSTALL_DIR}}/etc/newrelic-infra
    ARROW: '\033[0;36m===> \033[0;97m'

  tasks:
    default:
      cmds:
        - task: assert_pre_req
        - task: cleanup
        - task: setup_license
        - task: setup_proxy
        - task: install_infra
        - task: restart
        - task: assert_agent_started
        - task: assert_agent_status_ok

    assert_pre_req:
      cmds:
        - |
          echo {{.CONFIG_FILE}}
          echo {{.A_HOMEBREW_INSTALL_DIR}}
          echo {{.CONFIG_FILE_DIR}}
          exit 7
```

It returns the output:
```
DEBUG executing recipe infrastructure-agent-installer 
DEBUG /opt/homebrew/etc/newrelic-infra/newrelic-infra.yml 
DEBUG /opt/homebrew                                
DEBUG /opt/homebrew/etc/newrelic-infra             
DEBUG Task execution returned error   
```

Also I'm seeing that when brew is not installed, the message to offer installing it is not being printed :( 

Will you have some time to review this with me? 🙂 

Thanks!!

Reply:
Thank you for the detailed post! As agreed over Slack, I reached out to our hero (with more newrelic-cli experience) to help us here.